Tests/SwiftSyntaxMacrosTest/MacroSystemTests.swift

```diff
@@ -769,6 +889,70 @@ final class MacroSystemTests: XCTestCase {
     )
   }
 
+  func testPoundFunction() {
+    assertMacroExpansion(
```

Comment:
I would prefer to have a separate `assertMacroExpansion` invocation for each `#function`. In my experience that makes it a lot easier to 
a) check how `#function` is being transformed in each case and
b) debug the test case if something is going wrong